cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeService.java

```diff
@@ -343,6 +349,11 @@ public Void call() throws Exception {
                                                   String metricsConfPath, @Nullable String logbackJarName) {
     Map<String, String> configs = new HashMap<>();
 
+    // Make Spark UI runs on random port. By default, Spark UI runs on port 4040 and it will do a sequential search
+    // of the next port if 4040 is already occupied. However, during the process, it unnecessarily logs big stacktrace
+    // as WARN, which pollute the logs a lot if there are concurrent Spark job running (e.g. a fork in Workflow).
+    configs.put("spark.ui.port", "0");
```

Comment:
So if a user runs spark outside of cdap, the UI ports will be ``4040``, ``4041``, ``4042``... But when he runs them through CDAP, they will be random? Could this be a problem to users who want to use things like ``lsof`` to determine details about open ports (if they're used to things like ``lsof -i "tcp:404x"``? Should we worry about that? 

Will spark log the port on the console when the job runs?